hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/ActionStatusHandlerService.java

```diff
@@ -0,0 +1,90 @@
+package org.eclipse.hawkbit.repository.jpa;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import org.eclipse.hawkbit.im.authentication.SpPermission.SpringEvalExpressions;
+import org.eclipse.hawkbit.im.authentication.TenantAwareAuthenticationDetails;
+import org.eclipse.hawkbit.repository.ControllerManagement;
+import org.eclipse.hawkbit.repository.EntityFactory;
+import org.eclipse.hawkbit.repository.builder.ActionStatusCreate;
+import org.eclipse.hawkbit.repository.event.remote.TargetStatusForDistributionSetEvent;
+import org.eclipse.hawkbit.repository.model.Action;
+import org.eclipse.hawkbit.repository.model.Action.Status;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.event.EventListener;
+import org.springframework.security.authentication.AnonymousAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.context.SecurityContextImpl;
+
+/**
+ * A service that listens and processes the TargetStatusForDistributionSetEvent
+ *
+ */
+public class ActionStatusHandlerService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ActionStatusHandlerService.class);
+
+    private final ControllerManagement controllerManagement;
+    private final EntityFactory entityFactory;
+
+    public ActionStatusHandlerService(final ControllerManagement controllerManagement,
+            final EntityFactory entityFactory) {
+        this.controllerManagement = controllerManagement;
+        this.entityFactory = entityFactory;
+    }
+
+    @EventListener(classes = TargetStatusForDistributionSetEvent.class)
+    public void handle(final TargetStatusForDistributionSetEvent event) {
+        Action action = controllerManagement.findActiveActionsByTargetAndDistributionSet(event.getTargetId(),
+                event.getDistributionSetId());
```

Comment:
I am confused: `event.getTargetId` contains the `controllerId`? if yes maybe rename the event-property to `getTargetControllerId()` or the like